web/src/user_profile.ts

```diff
@@ -540,6 +545,67 @@ export function show_user_profile_access_error_modal(): void {
     });
 }
 
+function add_user_to_groups(group_ids: number[], user_id: number, $alert_box: JQuery): void {
+    let success_count = 0;
+
+    // Recursive function to process groups in series (one after another)
+    function add_user_to_next_group(index: number): void {
+        if (index >= group_ids.length) {
+            // If all groups are processed without errors, show a single success banner
+            if (success_count > 0) {
+                ui_report.success(
+                    $t_html({
+                        defaultMessage: "Successfully added to group(s)!",
+                    }),
+                    $alert_box,
+                    1200,
+                );
+            }
+            return;
+        }
+
+        const group_id = group_ids[index]!;
+        const target_user_group = user_groups.get_user_group_from_id(group_id);
+
+        if (!target_user_group) {
+            return;
+        }
+
+        user_group_edit_members.edit_user_group_membership({
+            group: target_user_group,
+            added: [user_id],
+            success(): void {
+                success_count += 1;
+
+                // Clear the pill for the successfully added group
+                const $pill = $(
+                    `#user-group-to-add .pill-container .pill[data-user-group-id="${target_user_group.id}"]`,
+                );
+                if ($pill.length > 0) {
+                    $pill.find(".exit").first().trigger("click");
+                }
```

Comment:
Instead of removing these pills one by one, would be better to just clear it once - either when user has been added to all groups successfully or after we get the first error.